src/Minimap.ts

```diff
@@ -0,0 +1,288 @@
+/**
+ * This is an extension adds support for adding a minimap to one of the map's control containers.
+ */
+
+import { Map } from "./Map";
+import { DOMcreate, DOMremove } from "./tools";
+
+import type {
+  ControlPosition,
+  FillLayerSpecification,
+  GeoJSONSource,
+  IControl,
+  LineLayerSpecification,
+  StyleOptions,
+  StyleSpecification,
+  StyleSwapOptions,
+} from "maplibre-gl";
+import type { MapOptions } from "./Map";
+import type { MapStyleVariant, ReferenceMapStyle } from "@maptiler/client";
+
+export interface ParentRect {
+  lineLayout: LineLayerSpecification["layout"];
+  linePaint: LineLayerSpecification["paint"];
+  fillPaint: FillLayerSpecification["paint"];
+}
+
+export interface MinimapOptionsInput {
+  /**
+   * Style of the map. Can be:
+   * - a full style URL (possibly with API key)
+   * - a shorthand with only the MapTIler style name (eg. `"streets-v2"`)
+   * - a longer form with the prefix `"maptiler://"` (eg. `"maptiler://streets-v2"`)
+   */
+  style?: ReferenceMapStyle | MapStyleVariant | StyleSpecification | string;
+
+  /**
+   * Set the zoom difference between the parent and the minimap
+   * If the parent is zoomed to 10 and the minimap is zoomed to 8, the zoomAdjust should be 2
+   * Default: -4
+   */
+  zoomAdjust?: number;
+
+  /** Set a zoom of the minimap and don't allow any future changes */
+  lockZoom?: number;
+
+  /** Adjust the pitch only if the user requests */
+  pitchAdjust?: boolean;
+
+  /** Set CSS properties of the container using object key-values */
+  containerStyle?: CSSStyleDeclaration & Record<string, string>;
+
+  /** Set the position of the minimap at either "top-left", "top-right", "bottom-left", or "bottom-right" */
+  position?: ControlPosition;
+
+  /** Set the parentRect fill and/or line options */
+  parentRect?: ParentRect;
+}
+
+export interface MinimapOptions extends MapOptions {
+  zoomAdjust: number;
+  pitchAdjust: boolean;
+  containerStyle: Record<string, string>;
+  parentRect?: ParentRect;
+}
+
+export default class Minimap implements IControl {
+  #options: MinimapOptions;
```

Comment:
From what I understand, you are using the `#` character to convey that a class property or method is private but using the TS term `private`:
- is much more explicit
- is the same as the other files from the SDK
- allow to double click on the property name to copy paste while the `#` would not be selected

Reply:
This is an interesting discussion. So in Typescript, the `private` keyword checks at compile time but the `#` **enforces** at compile time. This is an important distinction because certain variables and functions in the class I do not want acessable. The class itself is exported by the project, so someone could extend the class or use it in a particular way in which I had not intended.

There is a great discussion on the distinction [here](https://stackoverflow.com/a/59641565) and the proposal discusses it [here](https://github.com/tc39/proposal-class-fields#private-fields).

Ultimately, if you'd still prefer `private` that is fine too.

Reply:
Ok, interesting. i was not aware of this difference. I guess it's fine then